components/org.wso2.carbon.identity.organization.management.organization.user.sharing/src/main/java/org/wso2/carbon/identity/organization/management/organization/user/sharing/UserSharingPolicyHandlerServiceImpl.java

```
@@ -1006,4 +1124,15 @@ private ApplicationManagementService getApplicationManagementService() {
 
         return OrganizationUserSharingDataHolder.getInstance().getApplicationManagementService();
     }
+
+    private void startTenantFlowFromOrganization(String sharingInitiatedOrgId) {
+
+        try {
+            String sharingInitiatedTenantDomain = getOrganizationManager().resolveTenantDomain(sharingInitiatedOrgId);
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(sharingInitiatedTenantDomain, true);
+        } catch (OrganizationManagementException e) {
+            LOG.error("Error occurred while starting tenant flow from organization: " + sharingInitiatedOrgId, e);
```

throw exception instead logging the error

---

Since this is an asynchronous operation, directly throwing an exception could cause issues. For instance, `CompletableFuture` does not implicitly allow handled exceptions to be thrown. Additionally, since the API response has already been delivered at this point, there is no direct way to propagate the error back to the user. Therefore, logging the error would be more appropriate in this case. Moreover, we can later trigger event handlers to indicate that these operations have failed in a structured manner rather than relying solely on logging.